Improve documentation for `kubernetes.io/enforce-mountable-secrets` (#42741)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

Used on: ServiceAccount

The value for this annotation must be **true** to take effect.
When set to **true**, this annotation enforces the following rules for Pods running as this ServiceAccount:

1. Secrets mounted as volumes must be listed in the ServiceAccount's `secrets` field.
2. Secrets referenced in `envFrom` for containers or `initContainers` must also be listed in the ServiceAccount's `secrets` field.
3. Secrets referenced in a Pod's `imagePullSecrets` must be present in the ServiceAccount's `imagePullSecrets` field.

> While the annotation's name suggests it only concerns the mounting of secrets,

Reviewer: ```suggestion
While the annotation's name suggests it only concerns the mounting of Secrets,
```

Author: Thank you for your review.

its enforcement also extends to other ways Secrets are used in the context of a Pod.

> Therefore, it is crucial to ensure that all the referenced secrets are correctly specified in the ServiceAccount.

Reviewer: ```suggestion
Therefore, it is crucial to ensure that all the referenced Secrets are correctly specified in the ServiceAccount.
```

Reviewer 2: I would put this advice in the ServiceAccount page, not here. And strongly encourage readers to go look there.

Used on: ServiceAccount

The value for this annotation must be **true** to take effect.

> When set to "true", this annotation enforces the following rules for Pods running as this ServiceAccount:

Reviewer 2: ```suggestion
When you set this annotation  to "true", Kubernetes enforces the following rules for
Pods running as this ServiceAccount:
```

1. Secrets mounted as volumes must be listed in the ServiceAccount's `secrets` field.
1. Secrets referenced in `envFrom` for containers or `initContainers` must also
   be listed in the ServiceAccount's `secrets` field.
1. Secrets referenced in a Pod's `imagePullSecrets` must be present in the
   ServiceAccount's `imagePullSecrets` field.

> 

Reviewer 2: Maybe state what happens to a Pod that doesn't meet those requirements, and also when the enforcement happens.

Used on: ServiceAccount

The value for this annotation must be **true** to take effect.

> When set to **true**, this annotation enforces the following rules for Pods running as this ServiceAccount:

Reviewer: ```suggestion
When set to "true", this annotation enforces the following rules for Pods running
as this ServiceAccount:
```

> 1. Secrets mounted as volumes must be listed in the ServiceAccount's `secrets` field.
> 2. Secrets referenced in `envFrom` for containers or `initContainers` must also be listed in the ServiceAccount's `secrets` field.
> 3. Secrets referenced in a Pod's `imagePullSecrets` must be present in the ServiceAccount's `imagePullSecrets` field.

Reviewer: ```suggestion
1. Secrets mounted as volumes must be listed in the ServiceAccount's `secrets` field.
1. Secrets referenced in `envFrom` for containers or `initContainers` must also
   be listed in the ServiceAccount's `secrets` field.
1. Secrets referenced in a Pod's `imagePullSecrets` must be present in the
   ServiceAccount's `imagePullSecrets` field.
```

Used on: ServiceAccount

The value for this annotation must be **true** to take effect.
When you set this annotation  to "true", Kubernetes enforces the following rules for
Pods running as this ServiceAccount:

1. Secrets mounted as volumes must be listed in the ServiceAccount's `secrets` field.
1. Secrets referenced in envFrom for containers or initContainers must also be listed in the ServiceAccount's secrets field.

>    If an initContainer references a secret not listed in the ServiceAccount's secrets field,

Reviewer 2: Is this right?
```suggestion
   If any container in a Pod references a Secret not listed in the ServiceAccount's `secrets` field
   (and even if the reference is marked as `optional`), then
```

   the Pod will fail to start, and an error indicating the non-compliant secret reference will be generated.
1. Secrets referenced in a Pod's `imagePullSecrets` must be present in the
   ServiceAccount's `imagePullSecrets` field, the Pod will fail to start,
   and an error indicating the non-compliant image pull secret reference will be generated.

When you create or update a Pod, these rules are checked. If a Pod doesn't follow them, it won't start and you'll see an error message.

> 

Reviewer 2: ```suggestion
If a Pod is already running and you change the `kubernetes.io/enforce-mountable-secrets` annotation
to true, or you edit the associated ServiceAccount to remove the reference to a Secret
that the Pod is already using, the Pod continues to run.

```
(is this right?)
